FDS Review (#2665)
inductiva/inductiva: docs-sources/fds/scaling-with-mpi.md

# Scale FDS with MPI
Fire Dynamics Simulator (FDS) supports parallelism using two methods:
- **MPI (Message Passing Interface)**: Distributes the workload across multiple meshes by running separate MPI processes.

> - **OpenMP**: Enables multi-threading within each mesh.

Reviewer: This is repeated. Maybe we can move to FAQ.

Estimated computation cost (US$): 0.0057 US$
```

The simulation took approximately 21 minutes. Let's use MPI to make it faster.

## Running with Multiple MPI Processes
To run in parallel using MPI:
- Use an input file with multiple meshes (e.g., `strong_scaling_test_008.fds`)

> - Select a machine with at least as many virtual CPUs (vCPUs) as the number of MPI processes (for the 8-mesh case, `c2d-standard-8` is a good fit)

Author: it's not a tutorial tho, it's a benchmarks page

task.print_summary()
```

> > ⚠️ **Note on vCPUs and Hyperthreading**: In most cloud environments (like Google Cloud), a vCPU corresponds to a hyperthread, not a full physical core. So a `c2d-standard-8` machine has 8 vCPUs, which typically means 4 physical cores with hyperthreading enabled. In this setup, each MPI process runs on a separate hyperthread by default.

Reviewer: ```suggestion
> ⚠️ **Note on vCPUs and Hyperthreading**: In most cloud environments (like Google Cloud), a vCPU corresponds to a thread, not a full physical core. So a `c2d-standard-8` machine has 8 vCPUs, which typically means 4 physical cores with hyperthreading enabled.
```

## Results
The table below compares performance across different MPI configurations for the 1-, 8-, and 32-mesh cases. Speed-up is calculated relative to the 1-mesh baseline.

> | MPI Processes | Machine Type     | Execution Time  | Estimated Cost (USD) | Speed-up |

Reviewer: Lets keep the tables consistent with the others.

| 1             | c2d-standard-2   | 21 minutes, 15s | 0.0057               | 1.0×     |
| 8             | c2d-standard-8   | 5 minutes, 22s  | 0.0047               | 4.0×     |
| 32            | c2d-standard-32  | 1 minute, 49s   | 0.0062               | 11.7×    |

Increasing parallelism from 1 to 8 MPI processes results in a **4× speed-up**, reducing runtime from over 21 minutes to just over 5 minutes. This demonstrates a clear performance gain. However, due to typical overheads in parallel computing, such as communication and synchronization, the speed-up is not perfectly linear.

Scaling further from 8 to 32 processes yields an additional 2.9× speed-up, cutting total runtime to under 2 minutes — **over 11× faster** than the baseline.

While the simulation cost increases slightly with more resources, the main benefit is that you’ll get results much faster.

With Inductiva, you have full flexibility to choose the computational resources that best match your time and budget constraints.

> **Note**: The 1-process simulation appears more expensive because it used only 1 of the 2 available vCPUs. You’re billed for the full machine, regardless of how many cores are utilized.

> ## Advanced Setup: Disabling Hyperthreading

Reviewer: In the future we can point to the other document I created about hyper-threading.

Oh and it is hyper-threading. I also didn't know it. We need to fix that type all over our docs.

Reviewer: It is hyper-threading. Not hyperthreading.
